packages/zoe/src/contracts/priceAggregator.js

```diff
@@ -0,0 +1,334 @@
+
+  /** @type {Store<Instance, Set<OracleRecord>>} */
+  const instanceToRecords = makeStore('oracleInstance');
+
+  let publishedTimestamp = await E(timer).getCurrentTimestamp();
+
+  // Wake every POLL_INTERVAL and run the queriers.
+  const repeaterP = E(timer).createRepeater(0, POLL_INTERVAL);
+  /** @type {TimerWaker} */
+  const waker = {
+    async wake(timestamp) {
+      // Run all the queriers.
+      const querierPs = [];
+      oracleRecords.forEach(({ querier }) => {
+        if (querier) {
+          querierPs.push(querier(timestamp));
+        }
+      });
+      await Promise.all(querierPs);
+    },
+  };
+  E(repeaterP).schedule(waker);
+
+  /**
+   * @param {Object} param0
+   * @param {number} [param0.overrideBaseValueOut]
+   * @param {Timestamp} [param0.timestamp]
+   */
+  const makeCreateQuote = ({ overrideBaseValueOut, timestamp } = {}) =>
+    /**
+     * @param {PriceQuery} priceQuery
+     * @returns {ERef<PriceQuote>=}
+     */
+    function createQuote(priceQuery) {
+      // Sniff the current baseValueOut.
+      const baseValueOut =
+        overrideBaseValueOut === undefined
+          ? publishedBaseValueOut
+          : overrideBaseValueOut;
+      if (baseValueOut === undefined) {
+        return undefined;
+      }
+
+      /**
+       * @param {Amount} amountIn the given amountIn
+       * @returns {Amount} the amountOut that will be received
+       */
+      const calcAmountOut = amountIn => {
+        const valueIn = mathIn.getValue(amountIn);
+        return mathOut.make(
+          floorDivide(multiply(valueIn, baseValueOut), baseValueIn),
+        );
+      };
+
+      /**
+       * @param {Amount} amountOut the wanted amountOut
+       * @returns {Amount} the amountIn needed to give
+       */
+      const calcAmountIn = amountOut => {
+        const valueOut = mathOut.getValue(amountOut);
+        return mathIn.make(
+          ceilDivide(multiply(valueOut, baseValueIn), baseValueOut),
+        );
+      };
+
+      // Calculate the quote.
+      const quote = priceQuery(calcAmountOut, calcAmountIn);
+      if (!quote) {
+        return undefined;
+      }
+
+      const {
+        amountIn,
+        amountOut,
+        timestamp: theirTimestamp = timestamp,
+      } = quote;
+      mathIn.coerce(amountIn);
+      mathOut.coerce(amountOut);
+      if (theirTimestamp !== undefined) {
+        return authenticateQuote([
+          { amountIn, amountOut, timer, timestamp: theirTimestamp },
+        ]);
+      }
+      return E(timer)
+        .getCurrentTimestamp()
+        .then(now =>
+          authenticateQuote([{ amountIn, amountOut, timer, timestamp: now }]),
+        );
+    };
+
+  /**
+   * @param {Array<number>} samples
+   * @param {Timestamp} timestamp
+   */
+  const updateQuote = async (samples, timestamp) => {
+    const median = calculateMedian(
+      samples.filter(sample => sample > 0 && Number.isSafeInteger(sample)),
+      { add, divide: floorDivide, isGTE },
+    );
```

Comment:
Not the history, the set of latest samples (one from each oracle).

Reply:
Ah, that's much clearer. I didn't figure out that it's combining multiple price feeds. Is there a place to say that explicitly so the next person to read this will have an easier time?